app/javascript/lib/scp-api.js

```diff
@@ -72,6 +80,67 @@ export async function fetchAuthCode(mock=false) {
   return authCode
 }
 
+/**
+* Create user annotation
+*
+* A "user annotation" is a named object of arrays.  Each item has a label
+* (`name`) and cell names (`values`).  Signed-in users can create these
+* custom annotations in the Explore tab of the Study Overview page.
+*
+* See user-annotations.js for more context.
+*
+* @param {String} studyAccession Study accession, e.g. SCP123
+* @param {String} cluster Name of cluster, as defined at upload
+* @param {String} annotation Full annotation name, e.g. "CLUSTER--group--study"
+* @param {String} subsample Subsampling threshold, e.g. 100000
+* @param {String} userAnnotationName Name of new annotation
+* @param {Object} selections User selections for new annotation.
+*    Each selection has a label (`name`) and list of cell names (`values`).
+*    See `prepareForApi` in `user-annotations.js` for details.
+*/
+export async function createUserAnnotation(
+  studyAccession, cluster, annotation, subsample,
+  userAnnotationName, selections, mock=false
+) {
+  const init = configPost(mock)
+
+  init.body = JSON.stringify({
+    name: userAnnotationName,
+    user_data_arrays_attributes: selections,
+    cluster, annotation, subsample
+  })
+
+  const apiUrl = `/site/studies/${studyAccession}/user_annotation`
+  const [jsonOrResponse, perfTime] = await scpApi(apiUrl, init, mock)
+
+  let message = ''
+  let annotations = {}
+  let errorType = null
+
+  // Consider refactoring this when migrating user-annotations.js to
+  // React, so components share more error handling logic and UI
+  if (jsonOrResponse.ok === false) {
```

Comment:
I agree this should be refactored to share error handling logic, but given the scope of this ticket it's fine for now.